x/gov/keeper/vote.go

```diff
@@ -29,6 +29,15 @@ func (keeper Keeper) AddVote(ctx context.Context, proposalID uint64, voterAddr s
 		return err
 	}
 
+	// HV2: check on the length of the options. This should never fail as it is only used by `MsgServer.Vote`.
```

Comment:
If it should never fail because we control all callsites and none should provide inputs that make this fail, then it's maybe worth panicking instead of returning an error. This comment also applies to other sanity checks you added in `tally.go`.

Reply:
Good one. Added `panic`s for such cases [here](https://github.com/0xPolygon/cosmos-sdk/pull/15)